sdk/cognitiveservices/textanalytics/Azure.AI.TextAnalytics/src/TextAnalyticsClient.cs

```diff
@@ -1157,6 +1183,33 @@ public virtual Response<ExtractLinkedEntitiesResultCollection> ExtractEntityLink
         #endregion
 
         #region Common
+
+        private static Dictionary<string, int> CreateIdToIndexMap(IEnumerable<TextDocumentInput> inputs)
+        {
+            var map = new Dictionary<string, int>();
```

Comment:
As discussed offline, if this could be a lot of input - and to use a dictionary for now - it's probably worth using `inputs.Count()` to initialize the dictionary's capacity. The cost of rebucketing would outweigh this extra O(n) call, but would otherwise be unnecessary.

In fact, I wonder if you should actually optimize for the single input case (methods) above, i.e. don't even call into this (or have a separate helper). You can still output a collection, but not go through all the extra allocations you have here.